include/irr/asset/IAsset.h

```diff
@@ -173,14 +173,24 @@ class IAsset : virtual public core::IReferenceCounted
 
         virtual core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const = 0;
 
-		//! restore() is not recursive!
-		virtual bool restore(IAsset* _other)
+		bool restoreFromDummy(IAsset* _other, uint32_t _levelsBelow)
```

Comment:
needs default argument for `_levelsBelow`

Reply:
if only asset manager can call this (just how it is for `convertToDummyObject`) then dont make the function public?